### DISCUSSION TITLE
Storage control modes are not reflected (anymore) in the adapter

### DISCUSSION THREAD

**Discussion by mtbsteve** [WillCodeForCats/solaredge-modbus-multi#678] [General]
### Describe the bug

The battery storage control mode settings are different between the SE app and the ModbusMulti adapter. If I recall correctly, the options were correct in an earlier 2.x version of the adapter.

### Expected behavior

I would expect to have the same settings and configurations available including:
- maximize self consumption
- Time of use
- Backup only
- Manual control

### Screenshots

![Screenshot 2024-09-21 083914](https://github.com/user-attachments/assets/654874f5-f651-45ed-96cb-b32bf4aed8d2)
![IMG_1054](https://github.com/user-attachments/assets/e4680128-2a5f-4fac-8789-ee4049ebc7ae)

### Diagnostic File

[config_entry-solaredge_modbus_multi-01J337KP50FCMME92VDYZC92EY.json](https://github.com/user-attachments/files/17083176/config_entry-solaredge_modbus_multi-01J337KP50FCMME92VDYZC92EY.json)

### Home Assistant Version

24.9.2

### solaredge-modbus-multi Version

3.0.0

### Installation Type

HAOS

### Read the Instructions

- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] I have read the Known Issues https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues
- [X] This issue is not about a template error (do not report template errors, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by WillCodeForCats** [comment]
Diagnostic file looks fine to me.

There are no changes to control mode setting names in code.

**Reply by mtbsteve** [reply]
Problem is that the options listed in the drop down menu in HA (see screenshot) do not match the options provided by the inverter. 
How shall I set eg the time of use mode within Homeassistant?

**Reply by WillCodeForCats** [reply]
I don't have a battery so I can't comment on the app. But to me those do not look like the same setting group. Time of use is under storage control mode select entity.

Someone with a battery that has experience using these settings on a real system would be better to answer these questions.

**Reply by rolandsteinmeyer** [reply]
I can confirm that command mode settings are different between the SE app and the ModbusMulti adapter when using TOU. I have dynamic pricing with Tibber in germany and setting TOU via SE App works using EPEX Spot setting + base price per kWh. Battery is charging up from grid when price is low and discharging at high grid price taking price difference and charge loss into account. Amount of kWh that needs to be charged seems to be predicted quite good and leaves room for possible later solar surplus. But setting this via SE App is not reflected in Modbus output and it is also not possible to set back to TOU via Modbus when I change it to remote control + Solar Power only (off) while EV charging from grid so not to drain the SE battery. I think we will have to wait for new firmware from SE to get this sorted out.

**Reply by mtbsteve** [reply]
I rather think that SE is using some not yet documented Modbus registers to manage ToU. 
Anyway this feature is awesome and I can wait until it’s controllable via the modbus adapter.
